zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -946,7 +947,7 @@ public ProcessTxnResult processTxn(TxnHeader header, Record txn, boolean isSubTx
                                                  : Code.OK.intValue();
 
                             subtxn.setType(OpCode.error);
-                            record = new ErrorTxn(ec);
+                            record = new ErrorTxn(ec, "");
```

Comment:
You don't need to use the overloaded constructor if path is empty.

Reply:
jute only provides constructors below, no `ErrorTxn(int)`

```java
  public ErrorTxn() {
  }
  public ErrorTxn(
        int err,
        String path) {
    this.err=err;
    this.path=path;
  }
```

Reply:
You're right. Sorry I mixed it with `ErrorResult`.

Reply:
A question here maybe. Should we use `""` or `null` to stand for "no meaningful path"? Now I prefer the later since it explicitly represent "non value here".